lib/algos/list/delimiterMaster.js

```diff
@@ -165,14 +170,54 @@ class DelimiterMaster extends Delimiter {
         return super.filter(obj);
     }
 
+    allowDelimiterRangeSkip(key) {
+        if (!this.prefixKeySeen) {
+            return true;
+        }
+        // prefix = prefix, key = prefix/. Can skip since key will be part of commonPrefixes.
+        if (key[key.length - 1] === this.delimiter && key.length > this.prefix.length && key.startsWith(this.prefix)) {
+            return true;
+        }
+        const lastIdx = key.lastIndexOf(this.delimiter);
+        return lastIdx === - 1 || !(key.slice(0, lastIdx + 1) === this.prefix);
+    }
+
+    nextSkipRange(key) {
+        // const baseIndex = this.prefix ? this.prefix.length : 0;
+        // const delimiterIndex = key.indexOf(this.delimiter, baseIndex);
+        // if (delimiterIndex >= 0) {
+        //     const commonPrefix = getCommonPrefix(key, this.delimiter, baseIndex);
+        //     console.log('inside can add common', {key, commonPrefix, prefix: this.prefix});
+        //     if (this.CommonPrefixes.indexOf(commonPrefix) === -1
+        //             && this[this.nextContinueMarker] !== commonPrefix) {
+        //         return true;
+        //     }
+        // }
+        // return false;
+
+        const delimEnd = this.prefix && this.prefix.endsWith(this.delimiter); // prefix/
```

Comment:
Ideally we would precompute this and store it in a class member to save some CPU cycles